models/issue.go

```diff
@@ -1225,7 +1232,12 @@ func NewIssue(repo *Repository, issue *Issue, labelIDs []int64, assigneeIDs []in
 	}
 	sess.Close()
 
-	if err = NotifyWatchers(&Action{
+	return nil
+}
+
+// newIssuePostInsert performs issue creation operations that need to be separate transactions
+func newIssuePostInsert(issue *Issue, repo *Repository) error {
```

Comment:
I think we should move the events handle codes out of models at first. That's my PR #8217 did.

Reply:
OK; would you like to make this PR dependent of #8217 ?

Reply:
@guillep2k I think maybe you could review #8217 at first. 😅

Reply:
@lunny  Done